src/libraries/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -386,6 +386,64 @@ public void TrimExcess()
             }
         }
 
+        /// <summary>
+        /// Ensures that the capacity of this Queue is at least the specified <paramref name="capacity"/>.
+        /// </summary>
+        /// <param name="capacity">The minimum capacity to ensure</param>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), capacity, SR.ArgumentOutOfRange_NeedNonNegNum);
+            }
+
+            if (_array.Length < capacity)
+            {
+                int newCapacity = (int)((long)(_array.Length == 0 ? 1 : _array.Length) * (long)GrowFactor / 100);
+                while (newCapacity < capacity)
+                {
+                    newCapacity = (int)((long)newCapacity * (long)GrowFactor / 100);
+                }
+
+                // MaxArrayLength is defined in Array.MaxArrayLength and in gchelpers in CoreCLR.
+                // It represents the maximum number of elements that can be in an array where
+                // the size of the element is greater than one byte; a separate, slightly larger constant,
+                // is used when the size of the element is one.
+                const int MaxArrayLength = 0x7FEFFFFF;
+                if ((uint)newCapacity > MaxArrayLength)
+                {
+                    newCapacity = MaxArrayLength;
+                    if (newCapacity < capacity) newCapacity = capacity;
+                }
+
+                if (newCapacity < _array.Length + MinimumGrow)
```

Comment:
Would remove this and factor in `MinimumGrow` in the original calculation.

Reply:
Sorry I have not figure out how to remove this and ensure it can be more than minumumGrow, could you please give detail info ?

Reply:
See my comment in https://github.com/dotnet/runtime/pull/47149#discussion_r576265295

Reply:
Fixed based on comment.
From my initial understand, MinimumGrow is to make sure every attempt should increase no less than it. If that is true, maybe its definition will be broken after removing the last check:

> var queue = new Queue(1);
queue.EnsureCapacity(2);